packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -78,7013 +73,9090 @@
 #endif
 #endif
 
-#ifdef HAVE_ZOLTAN2_OMP
-#include <omp.h>
+// This macro could just be KOKKOS_HAVE_CUDA but preserving it here to
+// facilitate testing this on and off. When ZOLTAN2_MJ_USE_CUDA_KERNEL is
+// defined, the main kernel is run using a parallel_for -> parallel_for loop
+// with atomics, instead of parallel_reduce -> parallel_reduce which gives
+// better performance on host. This occurs because CUDA is more efficient with
+// atomics than host code. I understand ScatterView may be a mechanism to allow
+// a single code pipe-line which runs both reduction or atomic patterns but did
+// not investigate that yet.
+// TODO: Implement ScatterView and get rid of this macro.
+#ifdef KOKKOS_HAVE_CUDA
+#define ZOLTAN2_MJ_USE_CUDA_KERNEL // Atomic Atomic Loops
 #endif
 
+// TODO: This macro will be removed after some further profiling.
+// This option is being maintained to evaluate the perfomance of using floats
+// versus doubles for the reduction arrays in the main kernel of MJ.
+#define ZOLTAN2_MJ_USE_FLOAT_ARRAY_FOR_KERNEL
+
 #define LEAST_SIGNIFICANCE 0.0001
```

Comment:
Please don't pollute the global namespace with non-namespaced macros in a public header file.

Reply:
changed all macros to be ZOLTAN2 prefix.